### ISSUE TITLE
Suggestion: build/autoBuild and fields modification

### ISSUE THREAD

**Issue by elsassph** [macronauts/haxe-macro-workshop-2015#6]
Eg. using `Context.getBuildFields()` and inspecting, modifying, adding fields of a type.

**Comment by jasononeil** [comment]
Any suggestions for a real life example?  2 simple ideas I've had:

1) Something like [mcli](https://github.com/waneck/mcli) but easier. So, read the functions in the class, and create a `main()` that turns the class into a CLI tool.  We [started this idea](https://github.com/macronauts/haxe-macro-workshop-2015/blob/master/automatic_cli_tools/problem/Calculator.hx) but never finished it.

2) A class that reads an outside file (HTML, JSON, an Image etc) and creates class fields in response.  For example, read a template file and create a class with properties that render the template.

**Comment by kevinresol** [comment]
https://gist.github.com/kevinresol/86e2746ea8701060feb8

I used this macro to read a folder, and turn the file names into static vars.

For example it generates:

``` haxe
public static inline var assets__mypic1_png = 'assets/mypic1.png';
public static inline var assets__mypic2_png = 'assets/mypic2.png';
public static var all = ['assets/mypic1.png', 'assets/mypic2.png'];
```

**Comment by elsassph** [comment]
Yep good example. I did something recursive which was creating sub-types from a complex Json.
